microprofile/WFLY-19592_MicroProfile_Fault_Tolerance_4_1.adoc

```diff
@@ -0,0 +1,131 @@
+:idseparator:       -
+
+== Overview
+
+The goal is to integrate MicroProfile (MP) Fault Tolerance (FT) 4.1 specification from the MP Platform 7.0 to WildFly at default stability level.
+To certify functionality, the MP FT TCK is run as part of the WildFly testsuite.
+
+=== User Stories
+
+User wants to deploy an application using MicroProfile Fault Tolerance 4.1 from MicroProfile Platform 7.0.
+Users electing to employ MicroProfile Telemetry metrics will have MicroProfile Fault Tolerance metrics collected as well.
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/WFLY-19592[WFLY-19592 MicroProfile Fault Tolerance 4.1]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19588[WFLY-19588 Implement MicroProfile Platform 7] – MP Platform parent issue
+* https://issues.redhat.com/browse/WFLY-19590[WFLY-19590 MicroProfile Telemetry 2.0] – dependency of this issue for full Telemetry metrics integration
+
+// EAP Tracking issue:
+// * https://issues.redhat.com/browse/EAP7-2225[EAP7-2225 MicroProfile Fault Tolerance 4.1]
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly[WildFly]
+
+=== Other Interested Projects
+
+* https://github.com/eclipse/microprofile-fault-tolerance[Eclipse MicroProfile Fault Tolerance] – MP Specification and TCK sources
+* https://github.com/smallrye/smallrye-fault-tolerance[SmallRye Fault Tolerance] – implementation
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Upgrade MP FT specification modules to 4.1.
+* Upgrade SmallRye FT to 6.5.x.
+* Enable collecting metrics with MP Telemetry when the subsystem is available in the server configuration.
+
+=== Changed requirements
+
+* N/A
+
+=== Non-Requirements
+
+* N/A
+
+=== Future Work
+
+None.
+
+== Backwards Compatibility
+
+The specification and the implementation do not introduce any incompatibilities from the previous version (MP FT 4.0).
+
+=== Default Configuration
+
+There are no changes to the existing configuration.
+
+=== Importing Existing Configuration
+
+The subsystem does not have any configuration, so existing subsystem configuration is not affected.
+
+=== Deployments
+
+The specification and the implementation do not introduce any incompatibilities from the previous version relevant to deployments.
+
+=== Interoperability
+
+* N/A
+
+== Implementation Plan
+
+For the integration with Telemetry metrics to function,
+the integration of its https://issues.redhat.com/browse/WFLY-19590[subsystem and modules] is required.
+Implementation will be done in the created `mp-7-wip` https://github.com/wildfly/wildfly/tree/mp-7-wip[branch] in WildFly upstream repository.
+
+== Admin Clients
+
+* No effect on the existing clients, there are no model changes.
+
+== Security Considerations
+
+* No security implications of this integration.
+
+[[test_plan]]
+== Test Plan
+
+Integration tests
+
```

Comment:
@rhusar - During my previous review I forgot to mention that [the MicroProfile test suite](https://github.com/jboss-eap-qe/eap-microprofile-test-suite) needs to be listed here, which has been used so far, since the initial integration and for all the upgrades it has gone through.

Reply:
@fabiobrz Thanks, added that! This information should really be provided by QE as I don't have a list of QE test suites for every component and also, the RFE is still missing a QE contact.

Reply:
Thanks @rhusar. 
My reasoning for the request I made is that the MicroProfile test suite has been used since the initial integration of MicroProfile specs into WildFly as a way to validate the changes and to acknowledge pre-check for them to be merged into `main`. 
This is also reinforced by the shift-left activities that are currently ongoing, which involve such test suite directly.

I do agree on the fact that any additional information about tests which is outside the community scope will be provided by QE.

And yes, the feature is still missing a QE contact since we're aiming at completing sizing activities for all the MP 7 platform specs RFEs, first. This is required in order to know who will be involved.